atomisticparsers/gromacs/parser.py

```diff
@@ -774,12 +774,51 @@ def get_mdp_file(self):
 
         return self.get_gromacs_file(self.mdp_ext)
 
-    def get_gromacs_file(self, ext):
+    def get_gromacs_file(self, ext, return_all=False):
         files = [d for d in self._gromacs_files if d.endswith(ext)]
 
         if len(files) == 0:
             return ''
 
+        if return_all:
+            all_files = {}
+            contain_basename, no_basename_match, counts = [], [], []
+            for f in files:
+                # We assume that the matching trajectory file has (or starts with)
+                # the same basename as the log file, e.g., out.log would correspond to
+                # out.tpr and out.trr and out.edr (or out_some.tpr ...).
+                # These files are therefore given first (second... ) priority in the dictionary.
+                if f.rsplit('.', 1)[0] == self._basename:
```

Comment:
Can you add helper functions for each of these cases?
 - The first 2 just return os.path.join
 - The last one does the counting logic

This way you can minimize repeated code and also the code readability will be better, cause it will be more clear that below you do the same thing but don't concatenate them